src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Helpers/RemoveUnnecessaryImports/AbstractUnnecessaryImportsProvider.cs

```diff
@@ -26,15 +26,21 @@ public ImmutableArray<TSyntaxNode> GetUnnecessaryImports(SemanticModel model, Te
     public ImmutableArray<TSyntaxNode> GetUnnecessaryImports(
         SemanticModel model, TextSpan? span, Func<SyntaxNode, bool>? predicate, CancellationToken cancellationToken)
     {
-        if (span.HasValue)
-        {
-            // Bail out if there are no usings/imports in the filter span.
-            var node = model.SyntaxTree.FindNode(span, findInTrivia: false, getInnermostNodeForTie: false, cancellationToken);
-            if (node.FirstAncestorOrSelf<TSyntaxNode>() is null)
```

Comment:
this check was bogus.  TSyntaxNode is an ImportClause.  But the span is the full span of the ImportStatement in this case.

Switched from looking upwards to see if we're in an ImportClause to instead check the span entirely to see if it *intersects* with an ImportClause (like an ImportStatement would).

Reply:
TSyntaxNode is Using/Import directive syntax?